dev/com.ibm.ws.jpa_22_fat/test-applications/jpa22query/src/jpa22query/web/JPAQueryTestServlet.java

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package jpa22query.web;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import javax.annotation.Resource;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import javax.persistence.TypedQuery;
+import javax.servlet.annotation.WebServlet;
+import javax.transaction.UserTransaction;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+import jpa22query.entity.Employee;
+
+@SuppressWarnings("serial")
+@WebServlet(urlPatterns = "/TestJPA22Query")
+public class JPAQueryTestServlet extends FATServlet {
+    @PersistenceContext(unitName = "JPAPU")
+    private EntityManager em;
+
+    @Resource
+    private UserTransaction tx;
+
+    private static boolean tablesSetup = false; // Expecting non-parallel test execution
```

Comment:
I think I missed this one before, but this should be volatile.  This is a safe enough change that you can just make it after the build is complete without needing to do a re-build